### PR TITLE
Updated ZNY and ZWY borders

### PR DESCRIPTION
**Pull request by kmoberg**
This PR updates the New York mainland and oceanic borders and ensures that VATSPY will display New York Oceanic when online. 

Karl
NYARTCC Air Traffic Manager
atm@nyartcc.org

### CODE REVIEW

**Review by Connorseds2002 — CHANGES_REQUESTED** [review]
I am not sure if NY_CTR will be covering both sectors as extended vis rages would not work

**Comment by ajying** [comment]
> I am not sure if NY_CTR will be covering both sectors as extended vis rages would not work by NY_FSS will be doing it

Hi Connor,

I'm the facilitiy coordinator for ZWY (NY_FSS) and the one who authored the updates. Karl (the NY ATM) and I have been in discussion and testing the updates as we developed them. The data files as submitted in the pull request correclty display sector ownership for ZNY and ZWY.

We've also submitted an email to Nestor for project access / approval.

**Comment by kmoberg** [comment]
@Connorseds2002 What @ajying say is correct. We have tested and verified that it is indeed displaying correctly with this update.

**Comment by ajying** [comment]
I think the merge conflicts should be resolved now

**Comment by Connorseds2002** [comment]
> @Connorseds2002 What @ajying say is correct. We have tested and verified that it is indeed displaying correctly with this update.

I know where you were coming from but was not showing during CTP but worked when I changed it to the on in now

**Comment by kmoberg** [comment]
> > @Connorseds2002 What @ajying say is correct. We have tested and verified that it is indeed displaying correctly with this update.
> 
> I know where you were coming from but was not showing during CTP but worked when I changed it to the on in now

We knew it wasn’t working during CTP. What I’m telling you is that this PR also contains the files required to fix it and we have tested it that it is indeed working.

**Comment by Connorseds2002** [comment]
I am not trying to be rude but I could see the same thing being used on ganda and it workes the same way that KZNY that I updated @kmoberg

**Review by Adrian2k — CHANGES_REQUESTED** [review]
image

There are several thousand changes now in this PR, please make sure you update the current files when submitting changes.

**Comment by ajying** [comment]
> I am not trying to be rude but I could see the same thing being used on ganda and it workes the same way that KZNY that I updated @kmoberg

We're all trying to accomplish the same thing here. Connor, I appreciate you taking initiative to try to get things done, I think in this case there was just a miscommunication. We were actively working on updates to fix the NY Oceanic display issue along with additional airspace/sectorization display issues and rolled those all into 1 update/pull request. Something got lost in translation, but with this PR plus the merge conflcit resolution should fix all of them.

**Comment by ajying** [comment]
> image
> 
> There are several thousand changes now in this PR, please make sure you update the current files when submitting changes.

Hmm, I used the GitHub inline merge conflcit resolution tool to resolve the conflicts. There should only be a few hundred lines difference. I will take a look, might be a line termination issue.

**Comment by ajying** [comment]
> image
> 
> There are several thousand changes now in this PR, please make sure you update the current files when submitting changes.

I believe I've fixed it now, I'm seeing 565 changes in the FIRBoundaries.dat diff and 7 changes in the VATSpy.dat diff. It was a LF vs CRLF issue, GitHub changed them all to LFs.

File: FIRBoundaries.dat [hunk 1]
```diff
@@ -1188,10 +1194,12 @@ CZQX|1|1|8|45.000000|-63.000000|65.000000|-30.000000|55.000000|-46.500000
 48.493611|-8.00000
 45.000000|-8.000000
 45.000000|-30.000000
-CZQM|0|1|2|41.866667|-68.681703|48.500000|-56.051667|45.183334|-62.366685
+CZQM|0|1|4|41.866667|-68.681703|48.500000|-56.051667|45.183334|-62.366685
```
Comment (NelisV) [inline FIRBoundaries.dat:1197]: This is just a line, is this intended?
image

**Review by Connorseds2002 — APPROVED** [review]

File: FIRBoundaries.dat [hunk 2]
```diff
@@ -1109,10 +1109,12 @@ CZQX|1|0|13|45.000000|-63.000000|65.000000|-30.000000|55.000000|-46.500000
 63.500000|-55.666667
 63.500000|-39.000000
 61.000000|-30.000000
-CZQX|0|1|2|41.866667|-68.681703|48.500000|-56.051667|45.183334|-62.366685
+CZQX|0|1|4|41.866667|-68.681703|48.500000|-56.051667|45.183334|-62.366685
```
Comment (NelisV) [inline FIRBoundaries.dat:1112]: similar situation, just a line
image

**Review by NelisV — CHANGES_REQUESTED** [review]
Aside from the 2 rogue lines, looks good to me.

**Comment by ajying** [comment]
Yes, those 2 are the Moncton/Gander separators. I tried to change as little as possible (ie. only moving things that border ZNY/ZWY) and left other lines/internal boundaries as much the same as possible. CZQM and CZQX are set up a little strangely in that the main sector is drawn around both FIRs and they draw a line as a separator as opposed to drawing around their own FIR and setting the other FIR as an extension. I didn't want to unilaterally decide to change that for them so I just extended the divider line based on the LOA information I have between ZWY and CZQM/CZQX.

File: FIRBoundaries.dat (hunk 2 above)
Comment (kmoberg) [reply FIRBoundaries.dat:1112]: See Alex reply below.

**Comment by NelisV** [comment]
Check, makes sense from an operational standpoint, however for it to be correct geometry it's better not to use a polygon to draw a line. Layering them on top of each other is the correct way, shouldn't make a difference in how it looks. I can see the same issue exists in the old version, but turning them into proper polygons would be a nice improvement :) 

image

**Comment by ajying** [comment]
So CZQM and CZQX have non-trivial sector delegations that also involve CZUL. http://czqm.ca/Airspace.aspx

Although changing them to the proper polygons would fix one issue, I'm reluctant to do that. I think if those are going to be fixed, it'd be good to show the actual shelves and sector stratification while we do it and that should probably be its own project with input from those FIRs separate from the ZNY/ZWY updates.

**Comment by NelisV** [comment]
Fair point, let's do it in a separate PR.

**Review by NelisV — APPROVED** [review]

**Review by Connorseds2002 — APPROVED** [review]

**Review by kmoberg — COMMENTED** [review]
Has been resolved! Was a line-feed issue.

**Review by Adrian2k — APPROVED** [review]

**Pull request merged**